File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program98.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and return difference between frequency of 
//		small characters and frequency of capital characters.			
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCapital,iSmall;

	public:

		Solution(string str,int iCnt=0,int iCapital=0,int iSmall=0)
		{
			this->str=str;
		}

		int CountDifference()
		{
			for(iCnt=0;iCnt<str.length();iCnt++)
			{
				if(str[iCnt]>='A' && str[iCnt]<='Z')
				{
					iCapital++;
				}
				else
				{
					iSmall++;
				}
			}	
			return iSmall-iCapital;
		}
};

int main()
{
	string str;
	int iRet=0;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	iRet = sObj->CountDifference();
	cout<<"Diffrence is: "<<iRet<<"\n";

	delete sObj;
	return 0;

}

/*
OUTPUT:


Enter string: MarvellouS
Diffrence is: 6

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program98.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and return difference between frequency of 
//		small characters and frequency of capital characters.			
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt,iCapital,iSmall;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
		this.iCapital=0;
		this.iSmall=0;
	}

	int CountDifference()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)>='A' && str.charAt(iCnt)<='Z')
			{
				iCapital++;
			}
			else
			{
				iSmall++;
			}
			iCnt++;
		}
		
		return iSmall-iCapital;
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		iRet = obj.CountDifference();

		System.out.println("Difference is: "+iRet);
	}
}

/*
OUTPUT:
Enter the string: 
VishaL
Difference is: 2

*/
